### PR TITLE
injector: support extra allowed controller service accounts

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Loong Dai <loong.dai@intel.com>

# Description

## Issue reference

Please reference the issue this PR will close: #4680

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by daixiang0** [comment]
Also fix #4748

**Comment by daixiang0** [comment]
Now assuming the name of service account is unique, adding namespace option is not hard but makes config harder.

**Comment by daixiang0** [comment]
@artursouza do I need to post a PR only for fix #4748 and this will be merged in next release?

**Comment by daixiang0** [comment]
@yaron2 please take a review.

File: pkg/injector/config.go [hunk 1]
```diff
@@ -21,12 +21,13 @@ import (
...
-	SidecarImagePullPolicy string `envconfig:"SIDECAR_IMAGE_PULL_POLICY"`
-	Namespace              string `envconfig:"NAMESPACE" required:"true"`
-	KubeClusterDomain      string `envconfig:"KUBE_CLUSTER_DOMAIN"`
+	TLSCertFile                string `envconfig:"TLS_CERT_FILE" required:"true"`
+	TLSKeyFile                 string `envconfig:"TLS_KEY_FILE" required:"true"`
+	SidecarImage               string `envconfig:"SIDECAR_IMAGE" required:"true"`
+	SidecarImagePullPolicy     string `envconfig:"SIDECAR_IMAGE_PULL_POLICY"`
+	Namespace                  string `envconfig:"NAMESPACE" required:"true"`
+	KubeClusterDomain          string `envconfig:"KUBE_CLUSTER_DOMAIN"`
+	AllowedServiceAccountInfos string `envconfig:"ALLOWED_SERVICE_ACCOUNT_INFOS"`
```
Comment (yaron2) [inline pkg/injector/config.go:30]: I would change `ALLOWED_SERVICE_ACCOUNT_INFOS` to `ALLOWED_SERVICE_ACCOUNTS`.

File: pkg/injector/injector.go [hunk 2]
```diff
@@ -120,34 +121,37 @@ func NewInjector(authUIDs []string, config Config, daprClient scheme.Interface,
...
 }
 
-func getServiceAccount(ctx context.Context, kubeClient kubernetes.Interface, allowedControllersServiceAccount string) (string, error) {
+// getServiceAccount parses "service-account:namespace" k/v list and returns an array of UID.
+func getServiceAccount(ctx context.Context, kubeClient kubernetes.Interface, allowedServiceAcccountInfos []string) ([]string, error) {
 	ctxWithTimeout, cancel := context.WithTimeout(ctx, getKubernetesServiceAccountTimeoutSeconds*time.Second)
 	defer cancel()
 
-	sa, err := kubeClient.CoreV1().ServiceAccounts(metav1.NamespaceSystem).Get(ctxWithTimeout, allowedControllersServiceAccount, metav1.GetOptions{})
+	serviceaccounts, err := kubeClient.CoreV1().ServiceAccounts("").List(ctxWithTimeout, metav1.ListOptions{})
```
Comment (yaron2) [inline pkg/injector/injector.go:136]: @daixiang0 you probably need to add the `list` verb permission here: https://github.com/dapr/dapr/blob/5753fa83b7bcfde35b322a386d7dfdb2ea46371c/charts/dapr/charts/dapr_rbac/templates/ClusterRoleBinding.yaml#L37.

**Review by yaron2 — CHANGES_REQUESTED** [review]

File: pkg/injector/injector.go (hunk 2 above)
Comment (daixiang0) [reply pkg/injector/injector.go:136]: Thanks, I forget it.

File: pkg/injector/config.go (hunk 1 above)
Comment (daixiang0) [reply pkg/injector/config.go:30]: OK.

**Comment by yaron2** [comment]
@daixiang0 there's one failing test. once it's fixed we can merge, LGTM overall.

File: Makefile [hunk 3]
```diff
@@ -225,7 +225,7 @@ ifneq ($(ADDITIONAL_HELM_SET),)
 endif
 docker-deploy-k8s: check-docker-env check-arch
 	$(info Deploying ${DAPR_REGISTRY}/${RELEASE_NAME}:${DAPR_TAG} to the current K8S context...)
-	$(HELM) install \
+	$(HELM) install --debug\
```
Comment (yaron2) [inline Makefile:228]: Can this be removed?

Comment (daixiang0) [reply Makefile:228]: Sure, it is only for debug.

**Review by yaron2 — APPROVED** [review]

**Pull request merged**